packages/contracts-bedrock/test/actors/OutputBisectionActors.sol

```diff
@@ -0,0 +1,341 @@
+        uint256 numClaims = GAME.claimDataLen();
+
+        // Pre-allocate the `moves_` array to the maximum possible length. Test environment, so
+        // over-allocation is fine, and more easy to read than making a linked list in asm.
+        uint256 movesLen = 0;
+        moves_ = new Move[](numClaims - processedBuf);
+
+        for (uint256 i = processedBuf; i < numClaims; i++) {
+            // Grab the observed claim.
+            IOutputBisectionGame.ClaimData memory observed = getClaimData(i);
+
+            // Determine the direction of the next move to be taken.
+            (Direction moveDirection, Position movePos) = determineDirection(observed);
+
+            // Continue if there is no move to be taken against the observed claim.
+            if (moveDirection == Direction.Noop) continue;
+
+            if (movePos.depth() <= GAME.SPLIT_DEPTH()) {
+                // output bisection
+                moves_[movesLen++] = handleOutputBisectionMove(moveDirection, movePos, i);
+            } else if (movePos.depth() <= GAME.MAX_GAME_DEPTH()) {
+                // execution trace bisection
+                moves_[movesLen++] = handleExecutionTraceBisectionMove(moveDirection, movePos, i);
+            } else {
+                // instruction step
+                moves_[movesLen++] = handleStepMove(moveDirection, movePos, i);
+            }
+        }
+
+        // Update the length of the `moves_` array to the number of moves that were added.
+        assembly {
+            mstore(moves_, movesLen)
+        }
+
+        // Increment `processedBuf` by the number of claims processed, so that next time around,
+        // we don't attempt to process the same claims again.
+        processedBuf += numClaims - processedBuf;
+    }
+
+    ////////////////////////////////////////////////////////////////
+    //                          INTERNAL                          //
+    ////////////////////////////////////////////////////////////////
+
+    /// @dev Helper function to determine the direction of the next move to be taken.
+    function determineDirection(IOutputBisectionGame.ClaimData memory _claimData)
+        internal
+        view
+        returns (Direction direction_, Position movePos_)
+    {
+        if (_claimData.parentIndex == type(uint32).max) {
+            // If we agree with the parent claim and it is on a level we agree with, ignore it.
+            if (Claim.unwrap(claimAt(_claimData.position)) == Claim.unwrap(_claimData.claim)) {
+                return (Direction.Noop, Position.wrap(0));
+            }
+
+            // The parent claim is the root claim. We must attack if we disagree per the game rules.
+            direction_ = Direction.Attack;
+            movePos_ = _claimData.position.move(true);
+        } else {
+            // If the parent claim is not the root claim, check if we disagree with it and/or its grandparent
+            // to determine our next move.
```

Comment:
Unless this actor assumes the max depth is 4, it should consider every other parent claim up to root. Otherwise, we could end up supporting a poisoned great-great-grandparent with a valid grandchild.

Reply:
Yep - this does need to change. Addressing some tweaks to the actor in the followup PR :)

ATM, this portion shares the same logic from the old `FaultDisputeGame` `GamePlayer`.

Reply:
Sounds good.